source/containers.tex

```diff
@@ -15032,3 +15034,2423 @@
+
+\pnum
+\tcode{layout_stride} provides a layout mapping
+where the strides are user-defined.
+
+\begin{codeblock}
+namespace std {
+  template<class Extents>
+  class layout_stride::mapping {
+  public:
+    using extents_type = Extents;
+    using size_type = typename extents_type::size_type;
+    using rank_type = typename extents_type::rank_type;
+    using layout_type = layout_stride;
+
+  private:
+    static constexpr rank_type @\exposid{rank_}@ = extents_type::rank();    // \expos
+
+  public:
+    // \ref{mdspan.layout.stride.ctor}, constructors
+    constexpr mapping() noexcept = default;
+    constexpr mapping(const mapping&) noexcept = default;
+    template<class OtherSizeType>
+      constexpr mapping(const extents_type&, span<OtherSizeType, rank_>) noexcept;
+    template<class OtherSizeType>
+      constexpr mapping(const extents_type&, const array<OtherSizeType, rank_>&) noexcept;
+
+    template<class StridedLayoutMapping>
+      constexpr explicit(@\seebelow@) mapping(const StridedLayoutMapping&) noexcept;
+
+    constexpr mapping& operator=(const mapping&) noexcept = default;
+
+    // \ref{mdspan.layout.stride.obs}, observers
+    constexpr const extents_type& extents() const noexcept { return @\exposid{extents_}@; }
+    constexpr array<size_type, @\exposid{rank_}@> strides() const noexcept
+    { return @\exposid{strides_}@; }
+
+    constexpr size_type required_span_size() const noexcept;
+
+    template<class... Indices>
+      constexpr size_type operator()(Indices...) const noexcept;
+
+    static constexpr bool is_always_unique() noexcept { return true; }
+    static constexpr bool is_always_contiguous() noexcept { return false; }
+    static constexpr bool is_always_strided() noexcept { return true; }
+
+    static constexpr bool is_unique() noexcept { return true; }
+    constexpr bool is_contiguous() const noexcept;
+    static constexpr bool is_strided() noexcept { return true; }
+
+    constexpr size_type stride(rank_type i) const noexcept { return @\exposid{strides_}@[i]; }
+
+    template<class OtherMapping>
+      friend constexpr bool operator==(const mapping&, const OtherMapping&) noexcept;
+
+  private:
+    extents_type @\exposid{extents_}@{};                    // \expos
+    array<size_type, @\exposid{rank_}@> @\exposid{strides_}@{};         // \expos
+  };
+}
+\end{codeblock}
+
+\pnum
+If \tcode{Extents} is not a specialization of \tcode{extents},
+then the program is ill-formed.
+
+\pnum
+\tcode{layout_stride::mapping<E>} is a trivially copyable type
+that models \libconcept{regular} for each \tcode{E}.
+
+\rSec4[mdspan.layout.stride.expo]{Exposition-only helpers}
+
+\pnum
+Let \tcode{\exposid{REQUIRED-SPAN-SIZE}(e, strides)} be:
+\begin{itemize}
+\item
+1, if \tcode{e.rank() == 0} is \tcode{true}, otherwise
+\item
+0, if the size of the multidimensional index space \tcode{e} is 0, otherwise
+\item
+1 plus the sum of products of \tcode{(e.extent($r$) - 1)} and \tcode{(strides[$r$])}
```

Comment:
```suggestion
1 plus the sum of products of \tcode{e.extent($r$) - 1} and \tcode{strides[$r$]}
```

Reply:
I think I like the parens around the first expression, but the ones around the second one can go.

Reply:
@jensmaurer Agreed. I'll make the change directly. Thanks!